EIPS/eip-5437.md

```diff
@@ -0,0 +1,183 @@
+D8qHuOwxs8Fy9evmkwAVl51evLFNT532I4LK0zHSbF8MccZjpEFMSKwalKJn02Ml
+yTd+ljYLf8SKMOLVps8kc4VyMR1lz0PwSpKDFOmkC1LRURpM7UTtCK+/RFg1OLyQ
+SKBmdI37KQARAQABiQG8BBgBCAAmFiEEw5P/Y80WfpVNnGPK0lSTW7ndZa0FAmLz
+M2YCGwwFCRLMAwAACgkQ0lSTW7ndZa2oFgv8DAxHtRZchTvjxtdLhQEUSHt80JCQ
+zgHd7OUI9EU3K+oDj9AKtKZF1fqMlQoOskgBsLy/xpWwyhatv2ONLtHSjYDkZ7qs
+jsXshqpuvJ3X00Yn9PXG1Z1jKl7rzy2/0DnQ8aFP+gktfu2Oat4uIu4YSqRsVW/Z
+sbdTsW3T4E6Uf0qUKDf49mK3Y2nhTwY0YZqJnuQkSuUvpuM5a/4zSoaIRz+vSNjX
+MoXUIK/f8UnWABPm90OCptTMTzXCC1UXEHTNm6iBJThFiq3GeLZH+GnIola5KLO1
++YbsFEchLfLZ27pWGfIbyppvsuQmrHef+J3g6sXybOWDHVYr3Za1fzxQVIbwoIEe
+ndKG0bu7ZAi2b/c8uH/wHT5IvtfzHLeSTjDqG8UyLTnaDxHQZIE9JIzWSQ1DSoNC
+YrU7CQtL+/HRpiGFHfClaXln8VWkjnUvp+Fg1ZPtE1t/SKddZ7m29Hd9nzUc0OQW
+MOA+HDqgA3a9kWbQKSloORq4unft1eu/FCra
+=O6Bf
+-----END PGP PUBLIC KEY BLOCK-----
+```
+
+3. IF `setSecurityContact` is implemented and a call to it has succeeded in setting a new security
+contact, an event `SecurityContactChanged` MUST be emitted
+with the identical passed-in-parameters of `setSecurityContact`
+
+4. It's also RECOMMENDED that an on-chain security notify method `securityNotify`
+to implemented to receive security notice onchain. If it's implemented and a call
+has succeeded, it MUST emit an `OnSecurityNotification` with identical pass-in-parameter data.
+
+5. Compliant interface MUST implement [EIP-165](./eip-165.md).
+<!-- TODO: add EIP-165 interfaces. -->
+<!-- TODO also consider requiring/recommending implementing EIP-5629 ERC-interface detection. -->
+
+6. It's recommended to set a bounty policy via `bountyPolicy` method. The `id = 0` is preserved for a full overview, while other digits are used for different individual bounty policies. The returned
+string will be URI to content of bounty policies.
+No particular format of bounty policy is specified.
+
+## Rationale
+1. For simplicity we specify a simple GPG scheme with a given encryption scheme and assume email
+addresses to contact. It's possible in the future EIPs will introduce flexibility of specifying
+encyrption scheme or delivery methods.
+
+2. We choose to also include optional method `setSecurityContact` to set security contact,
+anticipating it might change such as due to expiration of encryption keys or change of
+security contact. It's RECOMMENDED to implement such method.
+
+3. We choose to explicitly make `securityNotify` as `payable` which allow implementers to
+set a bar of ethers to be pay/staked when reporting a security issue. It is RECOMMENDED for
+a bounty policy.
+
+4. We intentionally add some additional information for future expandability.
+e.g. for `bountyPolicy` the `extraData` returned COULD potentially be used
+as the [EIP-5247](./eip-5247.md) bytecodes to specify a smart contract execution proposal.
+To stay less opinionated we leave this to the implementers. Future EIPs.
+are welcomed to extend specifying these.
+
+## Backwards Compatibility
+Currently, existing solutions such as OpenZeppelin use PlainText in source code
```

Comment:
```suggestion
Currently, existing solutions such as OpenZeppelin use plaintext in source code
```
PlainText™ is a registered trademark of the NonExistent corporation. Used without permission for laughs.

(/j obviously)